### ISSUE TITLE
Cavern misplacement with Terraforged

### ISSUE THREAD

**Issue by Matthysse** [yungnickyoung/Better-Caves#50]
Hello!
This is a rare bug (found just twice) but I got this:
![2020-05-12_10 22 38](https://user-images.githubusercontent.com/32510957/81665254-c9a3e280-9440-11ea-9f80-8dcfea4bf44d.png)
![2020-05-12_11 19 15](https://user-images.githubusercontent.com/32510957/81696074-c0783d00-9463-11ea-91f9-92c49c016e4f.png)
![2020-05-12_11 19 32](https://user-images.githubusercontent.com/32510957/81696084-c2da9700-9463-11ea-9ded-9c679fce0ec8.png)

I have Biomes O plenty, quark and terraforges mods.

**Comment by yungnickyoung** [comment]
Hey, can you provide an example world seed and location for this?

**Comment by yungnickyoung** [comment]
Really bizarre. I suspect one (or both) of BOP/Terraforged changes biome placement without updating it in the chunk's biome cache. From the first picture, looks like they may also change the heightmap (which is what I use for surface detection in 1.14+) without updating it?

**Comment by ArMoReDfUrY** [comment]
if you have quark installed it will happen too.
i had Terraforged + BOP + Quark
Terra works with BOP fine as Terra takes full control of the generation, just adding modded biomes to its own, but adding Quark to the mix will get you what you have in images 2+3. found it a few times. 
image 1 looks like it may be Better Caves generation cutting into it.
Terra does change the heightmaps and a bunch of other customised parameters.
personally i feel terraforged needs a lot more work improving biome spread, structures, lakes, rivers and such. will be great when finished i'm sure.

**Comment by Matthysse** [comment]
Yes I think @ArMoReDfUrY is right. Sorry I can't give you the seed right now, maybe tomorrow.

**Comment by yungnickyoung** [comment]
Looking at these again - 2 and 3 are definitely quark megacaves, which are kinda notorious for generating in weird scenarios. 
I'm not 100% on the first one, but I think quark also adds large, low-ceiling rooms, like a horizontal ravine. Pretty sure that's what that is.

TL;DR it's all quark lol

**Comment by ArMoReDfUrY** [comment]
i'd say your right on that :)
